ReactApp/src/misc/api-fetch-setup.js

```diff
@@ -110,12 +89,30 @@ function corsMiddleware(options, next) {
 	return next(options);
 }
 
-function createHeadersMiddleware(authToken) {
+function apiPathModifierMiddleware(options, next) {
```

Comment:
This is a different approach to replacing the URLs for .com-connected sites. I'll update it so it catches any potential errors if the path does not meet the expected format before splitting, etc. But any initial thoughts are welcome!